packages/node/src/transports/base.ts

```diff
@@ -0,0 +1,108 @@
+import { Event, Payload, Response, Status, Transport, TransportOptions } from '@amplitude/types';
+
+import * as http from 'http';
+import * as https from 'https';
+import * as url from 'url';
+
+/**
+ * Internal used interface for typescript.
+ * @hidden
+ */
+export interface HTTPRequest {
+  /**
+   * Request wrapper
+   * @param options These are {@see TransportOptions}
+   * @param callback Callback when request is finished
+   */
+  request(
+    options: http.RequestOptions | https.RequestOptions | string | url.URL,
+    callback?: (res: http.IncomingMessage) => void,
+  ): http.ClientRequest;
+}
+
+/** Base Transport class implementation */
+export abstract class BaseTransport implements Transport {
+  /** The Agent used for corresponding transport */
+  public module?: HTTPRequest;
+
+  public event?: Event;
+
+  /** The Agent used for corresponding transport */
+  public client?: http.Agent | https.Agent;
+
+  /** Locks transport after receiving 429 response */
+  private _disabledUntil: Date = new Date(Date.now());
```

Comment:
While a whole app can be throttled it's much more common for individual device ids to be throttled instead. Event ordering is also only important for the events belonging to a particular device id (maybe user id as well? I'm not sure). So there's no need to restrict the number of inflight requests as long as they're not for the same device id.

Reply:
k, will take this out.